form-flow/src/main/kotlin/com/ritense/formflow/service/FormFlowService.kt

```diff
@@ -36,6 +42,36 @@ class FormFlowService(
         return formFlowDefinitionRepository.getById(formFlowDefinitionId)
     }
 
+    fun open(formFlowInstance: FormFlowInstance) {
+        val step = formFlowInstance.formFlowDefinition.getStepByKey(formFlowInstance.getCurrentStep().stepKey)
+
+        step.onOpen?.forEach { expression ->
+            val spelExpression = parseSpelExpression(expression)
+
+            if (spelExpression != null) {
+                try {
+                    spelExpression.value
+                } catch (e: RuntimeException) {
+                    throw FormFlowExpressionExecutionException(
+                        formFlowInstance.formFlowDefinition.id.key,
+                        expression,
+                        e
+                    )
+                }
+            }
+        }
+    }
+
+    fun parseSpelExpression(expression: String, context: ParserContext? = null): Expression? {
```

Comment:
I think this needs to move to the definition. Probably on FormFlowStep level.

Reply:
Both FormFlowStep and FormFlowNextStep are going to use this and perhaps even more in the future. Since it doesn't tie to one specific domain entity, its better to put it into a service

Reply:
I disagree with putting it in the service, as this is logic that does not require injection of any sort. I see two options:

1. Add a static method to a helper class. Not a huge fan of this, but it is an option.
2. Create a base class that FormFlowStep and FormFlowNextStep both extend, so the method is available to both of them.